qiskit_finance/data_providers/wikipedia_data_provider.py

```diff
@@ -34,7 +34,7 @@ class WikipediaDataProvider(BaseDataProvider):
     def __init__(
         self,
         token: str | None = None,
-        tickers: str | list[str] | None = None,
+        tickers: list[str] | None = None,
```

Comment:
I am not sure this is right "fix" for mypy issue. Below it checks whether its  list or not and has logic for string splitting as it supports a string there. When you changed type from Union you removed the # type comment from the self._tickers line that informed mypy. Otherwise it infers the type by that it first sees. You could just define self._tickers as the type str | list[str] (rather than use a type comment) which should work I believe.